docs/sources/flow/reference/components/prometheus.exporter.snowflake.md

```diff
@@ -0,0 +1,99 @@
+    username =     "USERNAME"
+    password =     "PASSWORD"
+    warehouse =    "WAREHOUSE"
+}
+```
+
+## Arguments
+
+The following arguments can be used to configure the exporter's behavior.
+Omitted fields take their default values.
+
+| Name           | Type     | Description                                           | Default          | Required |
+|----------------|----------|-------------------------------------------------------|------------------|----------|
+| `account_name` | `string` | The account to collect metrics for.                   |                  | yes      |
+| `username`     | `string` | The username for the user used when querying metrics. |                  | yes      |
+| `password`     | `secret` | The password for the user used when querying metrics. |                  | yes      |
+| `role`         | `string` | The role to use when querying metrics                 | `"ACCOUNTADMIN"` | no       |
+| `warehouse`    | `string` | The warehouse to use when querying metrics.           |                  | yes      |
+
+## Blocks
+
+The `Snowflake` component does not support any blocks, and is configured
+fully through arguments.
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components.
+
+| Name      | Type                | Description                                                  |
+|-----------|---------------------|--------------------------------------------------------------|
+| `targets` | `list(map(string))` | The targets that can be used to collect `snowflake` metrics. |
+
+For example, the `targets` can either be passed to a `prometheus.relabel`
+component to rewrite the metric's label set, or to a `prometheus.scrape`
+component that collects the exposed metrics.
+
+The exported targets will use the configured [in-memory traffic][] address
+specified by the [run command][].
+
+[in-memory traffic]: {{< relref "../../concepts/component_controller.md#in-memory-traffic" >}}
+[run command]: {{< relref "../cli/run.md" >}}
+
+## Component health
+
+`prometheus.exporter.snowflake` is only reported as unhealthy if given
+an invalid configuration. In those cases, exported fields retain their last
+healthy values.
+
+## Debug information
+
+`prometheus.exporter.snowflake` does not expose any component-specific
+debug information.
+
+## Debug metrics
+
+`prometheus.exporter.snowflake` does not expose any component-specific
+debug metrics.
+
+## Example
+
+This example uses a [`prometheus.scrape` component][scrape] to collect metrics
+from `prometheus.exporter.snowflake`:
+
+```river
+prometheus.exporter.snowflake "example" {
+    account_name = "XXXXXXX-YYYYYYY"
+    username     = "USERNAME"
+    password     = "PASSWORD"
+    warehouse    = "WAREHOUSE"
+}
+
+// Configure a prometheus.scrape component to collect snowflake metrics.
+prometheus.scrape "demo" {
+  targets    = prometheus.exporter.snowflake.example.targets
+  forward_to = [ /* ... */ ]
+}
```

Comment:
Examples should ideally be copy-and-paste-able, only needing to replace some values to work properly. Something like this should be sufficiently complete (with a formatting change as well):
```suggestion
prometheus.exporter.snowflake "example" {
  account_name = "XXXXXXX-YYYYYYY"
  username     = "USERNAME"
  password     = "PASSWORD"
  warehouse    = "WAREHOUSE"
}

// Configure a prometheus.scrape component to collect snowflake metrics.
prometheus.scrape "demo" {
  targets    = prometheus.exporter.snowflake.example.targets
  forward_to = [ prometheus.remote_write.default.receiver ]
}

prometheus.remote_write "default" {
  endpoint {
    url = "REMOTE_WRITE_URL"
  }
}
```